server/src/main/java/io/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -92,6 +93,16 @@
 public class DruidCoordinator
 {
   public static final String COORDINATOR_OWNER_NODE = "_COORDINATOR";
+
+  public static Comparator<DataSegment> SEGMENT_COMPARATOR = Comparators.inverse(new Comparator<DataSegment>()
+  {
+    @Override
+    public int compare(DataSegment lhs, DataSegment rhs)
+    {
+      int retVal = Comparators.intervalsByStartThenEnd().compare(lhs.getInterval(), rhs.getInterval());
+      return retVal != 0 ? retVal : lhs.compareTo(rhs);
```

Comment:
instead of adding a check here, you can chain comparators using `Ordering.from(<intervalcomparator>).compound(Ordering.natural())`